net/LogInViewController.m

```diff
@@ -217,11 +218,10 @@ - (IBAction)logInButton:(UIButton *)sender
         {
             NSLog(@"good!!!!");
             
-            __weak LogInViewController *weakSelf = self;
             dispatch_async(dispatch_get_main_queue(), ^
                        {
-                           weakSelf.mapDelegate.currentUser = resUser;
-                           [weakSelf.navigationController popViewControllerAnimated:YES];
+                           [CurrentItems sharedItems].user = resUser;
+                           [self.navigationController popViewControllerAnimated:YES];
```

Comment:
If it's about mappDelegate - i've already removed it. (by adding singleton)
of it is about : " __weak LogInViewController *weakSelf = self;"